resources/smpp/ra/src/main/java/org/restcomm/slee/resource/smpp/SmppSessionsImpl.java

```diff
@@ -495,4 +492,113 @@ public void fireUnknownThrowable(Throwable throwable) {
 
 	}
 
+	public class SenderThread extends Thread {
+
+		private Boolean running = true;
```

Comment:
consider declaring this as "volatile" is different threads are going to read/modify this var.